gunrock/graph/dynamic_graph/slabhash_graph_base.cuh

```diff
@@ -42,28 +42,30 @@ struct SlabHashGraphBase {
   HashContextT* h_hash_context;
   HashContextT* d_hash_context;
 
-  std::vector<SizeT> buckets_per_table;
+  SizeT* buckets_per_table;
 
   SizeT* d_edges_per_node;
   int8_t* d_base_slabs;
 
   SizeT* d_edges_per_bucket;
   SizeT* d_buckets_offset;
 
+  SlabHashGraphBase(){};
   // todo: add node values
   /**
    * @brief Allocate maximum capacity memory for SlabHash graph.
    *
    * @param[in] max_nodes Maximum number of nodes that the graph can store
    * @param[in] max_buckets Maximum number of buckets that the graph will use
    */
-  SlabHashGraphBase(SizeT max_nodes = 1 << 20, SizeT max_buckets = 1 << 20) {
+  void Allocate(SizeT max_nodes = 1 << 20, SizeT max_buckets = 1 << 20) {
```

Comment:
Consider replacing the magic values `1 << 20`

Reply:
Once we have an application, I'll add these two numbers as optional command line parameters. But I have to set them somewhere for now.